internal/usecase/client/pendo/pendo_client_test.go

```diff
@@ -0,0 +1,374 @@
+		visitors     map[string]map[string]any
+		storeMetrics map[string]any
+	)
+	respBuilder := builder_pendo.NewSetMetadataResponse().
+		WithTotal(int64(len(metrics))).
+		WithKind(kind)
+	require.NotNil(t, respBuilder)
+	for i := range metrics {
+		if groups, ok = store[string(kind)]; !ok {
+			respBuilder.AddMissing(metrics[i].VisitorID)
+			respBuilder.IncFailed()
+			continue
+		}
+		if visitors, ok = groups[string(group)]; !ok {
+			respBuilder.AddMissing(metrics[i].VisitorID)
+			respBuilder.IncFailed()
+			continue
+		}
+		if storeMetrics, ok = visitors[metrics[i].VisitorID]; !ok {
+			respBuilder.AddMissing(metrics[i].VisitorID)
+			respBuilder.IncFailed()
+			continue
+		}
+		respBuilder.IncUpdated()
+		for k, v := range metrics[i].Values {
+			storeMetrics[k] = v
+		}
+	}
+
+	resp := respBuilder.Build()
+	require.NotNil(t, resp)
+	return resp
+}
+
+func TestSetMetadata(t *testing.T) {
+	kind := pendo.KindAccount
+	group := pendo.Group("custom")
+	var store PendoHash = PendoHash{
+		string(kind): {
+			string(group) /* group */ : {
+				"test-visitor-id" /* visitorId */ : {
+					"test-field-1" /* fieldName */ : 4,                 /* value */
+					"test-field-2" /* fieldName */ : "someStringValue", /* value */
+				},
+			},
+		},
+	}
+	// https://hassansin.github.io/Unit-Testing-http-client-in-Go#2-by-replacing-httptransport
+	cfg := helperPendoConfig()
+	client := helperNewPendo(cfg, func(req *http.Request) *http.Response {
+		metrics := helperSetMetadataPrepareRequest(t, req, kind, group)
+		resp := helperSetMetadataPrepareResponse(t, metrics, kind, group, store)
+		bytesResp, err := json.Marshal(resp)
+		require.NoError(t, err)
+		require.NotNil(t, bytesResp)
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(bytes.NewBuffer(bytesResp)),
+			Header:     make(http.Header),
+		}
+	})
+
+	// Panic when context is nil
+	assert.PanicsWithValue(t, "'ctx' is nil", func() {
+		client.SetMetadata(nil, "", "", nil)
+	})
+
+	// Error when wrong argument
+	ctx := app_context.CtxWithLog(context.TODO(), slog.Default())
+	resp, err := client.SetMetadata(ctx, "", "", nil)
+	require.EqualError(t, err, "'kind' is an empty string")
+	require.Nil(t, resp)
+
+	// Success
+	metrics := make(pendo.SetMetadataRequest, 0, 1)
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "test-visitor-id",
+		Values: map[string]any{
+			"test-field-1": 2,
+			"test-field-2": "anotherString",
+		},
+	})
+	metrics = append(metrics, pendo.SetMetadataDetailsRequest{
+		VisitorID: "thisVisitorDoesNotExist",
+	})
+
+	// Call the data
+	resp, err = client.SetMetadata(ctx, kind, group, metrics)
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+	assert.Equal(t, int64(2), resp.Total)
+	assert.Equal(t, int64(1), resp.Failed)
+	assert.Equal(t, int64(1), resp.Updated)
+	assert.Equal(t, []string{"thisVisitorDoesNotExist"}, resp.Missing)
+}
+
+func TestSetMetadataErrorNilTransport(t *testing.T) {
```

Comment:
I don't understand the purpose of this test as it doesn't seem to test pendo_client but rather the test code itself.

Reply:
Thanks! Renaming the test and adding additional information.

The goal of the test was to cover the path:

```
	// Launch request
	resp, err := c.Client.Do(req)
	switch {
	case err != nil:
		logger.Error("doing request to pendo service")
		return nil, err
```

And the only way I found to do that was returning nil from the transport helper function.